cloud/userdata.go

```diff
@@ -287,7 +300,13 @@ func makeUserData(ctx context.Context, env evergreen.Environment, h *host.Host,
 		return "", errors.Wrap(err, "error creating user data with multiple parts")
 	}
 
-	return multipartUserData, errors.Wrap(h.SaveJasperCredentials(ctx, env, creds), "problem saving Jasper credentials to host")
+	if creds != nil {
```

Comment:
Can we put this repeated block above the if statement on 272 to avoid the repeated code?

Reply:
I don't want to move it because this should only happen if we're positive that the script was actually properly created, and the behavior between merging/not merging user data parts can't be combined.